File: python/liczby23.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  liczby23.py

def liczby2():
    """
    Funkcja drukuje wszystkie liczby 2-cyfrowe,
    w których cyfry nie powtarzają się. Funckja
    zwraca ich liczbę. Wykluczone liczby: 11, 22, 33 itd.
    """
    licznik = 0
    for i in range(1, 10):
        for j in range(0, 10):
            if i!=j:
                print(" {}{} ".format(i, j), end="")
                licznik += 1
    return licznik 
      
def liczby2v2():
    licznik = 0
    for i in range(10, 99):
        liczba = i % 10
        if liczba != int(i / 10):
            print(i, " ", end="")
            licznik += 1
    return licznik

def liczby3():
    licznik = 0
    for i in range(1,10):
        for j in range(10):
            for k in range(10):
                if i!=j and i!=k and k!=j:
                    print(" {}{}{} ".format(i, j, k), end="")
                    licznik += 1
    return licznik
    
def main(args):
    print("Liczb 2-cyfrowych:", liczby2())
    print()
    print("Liczb 2-cyfrowych:", liczby2v2())
    print()
    print("Liczb 3-cyfrowych:", liczby3())
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/liczby23.cpp
/*
 * liczby23.cpp
 */


#include <iostream>

using namespace std;

int liczby2(){
    int licznik = 0;
    for(int i = 1; i <= 9; i++){
        for(int j = 0; j <= 9; j++){
            if(i!=j){
               cout << i << j << " ";
               licznik++; 
            }
        }
    }
    cout << endl;
    return licznik;
}

int liczby3(){
    int licznik = 0;
    for(int i=1; i<10; i++){
        for(int j=0; j<10; j++){
            for(int k=0; k<10; k++){
                if(i!=j && i!=k && k!=j){
                    cout << i << j << k << " ";
                    licznik++;
                }
            } 
        }
    }
    cout << endl;
    return licznik;
}

int main(int argc, char **argv){
	int wynik = liczby2();
    cout << "Liczb 2-cyfrowych: " << wynik << endl;
    wynik = liczby3();
    cout << "Liczb 3-cyfrowych: " << wynik << endl;
	return 0;
}

